### DISCUSSION TITLE
reusing value from earlier response not working

### DISCUSSION THREAD

**Discussion by LindevdH** [AnWeber/vscode-httpyac#167] [Q&A] [answered]
Good day,

In my first GET request named "Basics" the response is like:
```
[
   {
      "item1": "testvalue",
     "item2": "123456"
   }
]
```
In my second get request I want to reuse the value of "Basics" item  "item2".
If I am correct then the request should be build up like:
```
###
# @name Basics
get http://10.2.9.20:6548/getValues

###
# @name refered
# @ref Basics
get http://10.2.9.20:6548/getValuesLinked/{{Basics.item2}}

```
But instead I get the error:
	"message": "Incorrect request variable reference syntax, it should be \"variableName.(response|request).(headers|body).
What might be relevant is that the " @ref " does not light up in Blue but in green.

It would be nice if I could get this setup correctly and working. It would save me a lot of time.

**Comment by AnWeber** [comment, marked as answer]
@LindevdH The error is not generated by httpyac but by [vscode-restclient](https://github.com/Huachao/vscode-restclient/blob/42d2927efecdda1db8df4b491e9074932afe6246/src/models/httpVariableResolveResult.ts#L13) (you used CodeLens Send Request). To some extent you can probably solve it with vscode-restclient, even though `@ref` is not supported (search jsonpath in readme).
If you want to try your luck with my extension, it's best to uninstall vscode-restclient to avoid misunderstandings and use Code Lens `Send` or command `httpyac.send`. We have different approaches in detail. Your example should actually work, except that the array access is not respected. 

```
###
# @name Basics
get http://10.2.9.20:6548/getValues

###
# @name refered
# @ref Basics
get http://10.2.9.20:6548/getValuesLinked/{{Basics[0].item2}}
```

**Reply by LindevdH** [reply]
Thanks for the reply @AnWeber 
To get it work I hat to do a clean install of VSCode.To find out what the real problem was.
But It is working as described and needed. Thanks for the quick response.
